### PR TITLE
Bump sphinx to 3.0

### PR DESCRIPTION
**Pull request by eric-wieser**
Note that m2r is not compatible with this - my hope is we didn't need it anyway.

### CODE REVIEW

**Comment by utensil** [comment]
Oh no, we do need `m2r`......

**Comment by utensil** [comment]
`m2r` is introduced so `README.md` can be rendered correctly. Compare to https://galgebra.readthedocs.io/en/latest/ , the PR build ( https://galgebra--240.org.readthedocs.build/en/240/ )  has lost `README.md` completely......

**Comment by eric-wieser** [comment]
Is the readme the only .md file that is rendered?

Thanks for finding that!

**Comment by utensil** [comment]
> Is the readme the only .md file that is rendered?

Yes. To be honest, Sphinx caused me so much trouble for properly handling `.md` and `pandoc` was not doing great for converting `.tex` to `.rst` as well.

**Comment by eric-wieser** [comment]
Seems to work now, decided to monkey-patch `sphinx` to keep `m2r` working.

File: doc/conf.py [hunk 1]
```diff
@@ -51,6 +52,24 @@
...
+    return decorator
+
+# workaround until https://github.com/miyakogi/m2r/pull/55 is merged
+@monkeypatch(sphinx.registry.SphinxComponentRegistry)
+def add_source_parser(_old_add_source_parser, self, *args, **kwargs):
+    # signature is (parser: Type[Parser], **kwargs), but m2r expects
+    # the removed (str, parser: Type[Parser], **kwargs).
+    if isinstance(args[0], str):
+        args = args[1:]
+    return _old_add_source_parser(self, *args, **kwargs)
```
Comment (utensil) [inline doc/conf.py:71]: 👍

File: doc/conf.py [hunk 2]
```diff
@@ -95,7 +114,7 @@
 # You can specify multiple suffix as a list of string:
 #
 # source_suffix = ['.rst', '.md']
-source_suffix = ['.rst', '.md']
+source_suffix = ['.rst']
```
Comment (utensil) [inline doc/conf.py:117]: I'm surprised that it still works after `.md` is removed 😱

Comment (eric-wieser) [reply doc/conf.py:117]: I didn't even mean to do that...

Comment (eric-wieser) [reply doc/conf.py:117]: But the `README` is included via `includemd` anyway

Comment (utensil) [reply doc/conf.py:117]: Where is `includemd`?

Comment (eric-wieser) [reply doc/conf.py:117]: https://github.com/pygae/galgebra/blob/77320a5f8d0bc35c89c3d7eb81e6746193ca5049/doc/index.rst#L1

**Pull request merged**